iterators.tex

```diff
@@ -7402,6 +7674,130 @@
 \end{itemdecl}
 
 \begin{itemdescr}
-\pnum \returns \tcode{size(r)}
+\pnum \returns \tcode{\newtxt{ranges::}size(r)}
 \end{itemdescr}
+
+\rSec2[range.primitives.size]{\tcode{size}}
+\pnum
+The name \tcode{size} denotes a customization point object
+(\ref{customization.point.object}). The expression
+\tcode{ranges::size(E)} for a subexpression \tcode{E} with type
+\tcode{T} has the same effects as:
+
+\begin{itemize}
+\item
+  \tcode{extent<T>::value} if \tcode{T} is an array type.
+
+\item
+  Otherwise, \tcode{((const T\&)(E)).size()} if it is a valid expression whose
+  type \tcode{I} satisfies \tcode{Integral<I>()}.
+
+\item
+  Otherwise, \tcode{size((const T\&)(E))} if it is a valid expression
+  whose type \tcode{I} satisfies \tcode{Integral<I>()} when overload
+  resolution is performed in a context that includes the declaration
+  \tcode{void size(const auto\&) = delete;} and does not include
+  a declaration of \tcode{ranges::size}.
+
+\item
+  Otherwise,
+  \tcode{ranges::cend(E) - ranges::cbegin(E)}, except that \tcode{E}
+  is only evaluated once, if it is a valid expression and the types
+  of \tcode{ranges::cbegin(E)} and \tcode{ranges::cend(E)} meet the
+  syntactic requirements of \tcode{SizedIteratorRange}. If
```

Comment:
OK. IMO, we should eventually banish unsigned integer types from STL2, but now is not the time.